app/controllers/users.rb

```diff
@@ -0,0 +1,19 @@
+get '/users/new' do
+  @user = User.new
+  erb :"users/new"
+end
+
+post '/users' do
+  @user = User.create(:email => params[:email],
+                     :password => params[:password],
+                     :password_confirmation => params[:password_confirmation],
+                     :name => params[:name],
+                     :username => params[:username])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect to('/')
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :"users/new"
+    end
+end
```

Comment:
In post '/users' we don't need to create @user, just user is enough because we are not using this user object in the view, so if we don't use it we don't want to send unnecessary data to the view.